src/main/java/com/hivemq/extensions/services/auth/SecurityRegistryImpl.java

```diff
@@ -35,22 +37,31 @@
 
     private final @NotNull Authenticators authenticators;
     private final @NotNull Authorizers authorizers;
+    private final @NotNull HiveMQExtensions hiveMQExtensions;
 
     @Inject
-    public SecurityRegistryImpl(final @NotNull Authenticators authenticators, final @NotNull Authorizers authorizers) {
+    public SecurityRegistryImpl(
+            final @NotNull Authenticators authenticators,
+            final @NotNull Authorizers authorizers,
+            final @NotNull HiveMQExtensions hiveMQExtensions) {
         this.authenticators = authenticators;
         this.authorizers = authorizers;
+        this.hiveMQExtensions = hiveMQExtensions;
     }
 
     @Override
     public void setAuthenticatorProvider(final @NotNull AuthenticatorProvider authenticatorProvider) {
         checkNotNull(authenticatorProvider, "authenticatorProvider must not be null");
 
-        final IsolatedPluginClassloader classLoader =
-                (IsolatedPluginClassloader) authenticatorProvider.getClass().getClassLoader();
+        final HiveMQExtension extension =
+                hiveMQExtensions.getExtensionForClassloader(authenticatorProvider.getClass().getClassLoader());
+
+        if(extension == null || extension.getExtensionClassloader() == null){
+            return;
```

Comment:
Shouldn't this throw an IllegalStateException?

Reply:
no as we just dont care if an extension is disabled.